pipeline/mask_handler.py

```diff
@@ -24,54 +25,72 @@ def mask_handler(args):
+
+    with urllib.request.urlopen(url, timeout=timeout_seconds) as response:
+        with open("temp.fits", 'w+b') as f:
+            f.write(response.read())
+    nhits_nominal = hp.ud_grade(hp.read_map("temp.fits"), meta.nside, power=-2)
+    os.remove("temp.fits")
+
+    # If we don't use a custom nhits map, work with the nominal nhits map.
+    if not meta.use_input_nhits:
+        print("Using nominal apodized mask for analysis")
+        hp.write_map(meta.nhits_map_name, nhits_nominal, dtype=np.float32,
+                     overwrite=True)
+        nhits = nhits_nominal
+    else:
+        print("Using custom apodized mask for analysis")
+        nhits = meta.read_hitmap()
+    meta.timer.stop("nhits", "Get hits map", args.verbose)
+
     if args.plots:
         plt.figure(figsize=(16, 9))
         hp.mollview(nhits, cmap=cmap, cbar=False)
         hp.graticule()
-        plt.savefig(meta.binary_mask_name.replace('.fits', '_nhits.png'))
+        plt.savefig(meta.nhits_map_name.replace('.fits', '.png'))
 
     # Generate binary survey mask from the hits map
-    meta.timer.start("Computing binary mask")
-    nhits = hp.read_map(nhits_file)
+    meta.timer.start("binary")
     binary_mask = get_binary_mask_from_nhits(nhits, meta.nside,
                                              zero_threshold=1e-3)
     meta.save_mask("binary", binary_mask, overwrite=True)
-    meta.timer.stop("Computing binary mask", args.verbose)
+    meta.timer.stop("binary", "Computing binary mask", args.verbose)
 
     if args.plots:
         plt.figure(figsize=(16, 9))
         hp.mollview(binary_mask, cmap=cmap, cbar=False)
         hp.graticule()
         plt.savefig(meta.binary_mask_name.replace('.fits', '.png'))
 
-    if not args.self_assemble:
-        # Download SAT apodized mask used in the SO BB
-        # pipeline paper (https://arxiv.org/abs/2302.04276)
-        # and use as analysis mask
-        print("Download and save SAT apodized mask ...")
-        sat_apo_file = meta._get_analysis_mask_name()
-        urlpref = "https://portal.nersc.gov/cfs/sobs/users/so_bb/"
-        url = f"{urlpref}apodized_mask_bbpipe_paper.fits"
-        with urllib.request.urlopen(url, timeout=timeout_seconds):
-            urllib.request.urlretrieve(url, filename=sat_apo_file)
-            sat_apo_mask = hp.read_map(sat_apo_file, field=0)
-            sat_apo_mask = hp.ud_grade(sat_apo_mask, meta.nside)
-            meta.save_mask("analysis", sat_apo_mask, overwrite=True)
+    # Make nominal apodized mask from the nominal hits map
+    meta.timer.start("apodized")
+    nominal_mask = get_apodized_mask_from_nhits(
+        nhits_nominal, meta.nside,
+        galactic_mask=None,
+        point_source_mask=None,
+        zero_threshold=1e-3, apod_radius=meta.masks["apod_radius"],
+        apod_radius_point_source=meta.masks["apod_radius_point_source"],
+        apod_type="C1"
+    )
+    first_nom, second_nom = get_spin_derivatives(nominal_mask)
```

Comment:
In the future we might want to use pre-computed thresholds provided by an external script.

Reply:
Good point. Let's leave it as is for now but make optimal mask apodization a hack project for the near future!